nni/compression/pytorch/compressor.py

```diff
@@ -793,25 +799,54 @@ def find_conv_bn_patterns(self, model, dummy_input):
                     if successor.op_type == 'BatchNorm2d':
                         self.conv_bn_patterns[node_group.name] = successor.name
 
-    def step_with_optimizer(self):
-        pass
+    def record_shape(self, model, dummy_input):
```

Comment:
It seems to be an universal function which is not specific for quantizer. Maybe we should put it into other places like utils.py?

Reply:
There exists compressor-specific util function (like `get_modules_to_compress`) and currently it is only used by quantization. Putting this function as a attribute function of quantizer may be a good choice.. :)